### DISCUSSION TITLE
Reusing value of a date field in another date field

### DISCUSSION THREAD

**Discussion by SSDK-Hello** [team-forms/community#7] [Q&A] [answered]
How can I reuse the value of a date field as a default value for another date field,

**Comment by Erin-Dwyer** [comment, marked as answer]
Hi @SSDK-Hello 

It sounds like you would like to automatically populate the value of a second date/time field based on the value of an initial date/time field. There a multiple ways you can achieve this:

[Demo](https://share.teamforms.app/form/NDNhODU0YTEtOTE2MS00MjQ3LWIzNTQtZWMxMWIzOTc1NDFiOjNlMDFmODQ2LTdhZTMtNDIyYi04ZWU3LTVhNTY1MTQxOTNiZTozNmZiMGU4My1hZjhhLTRkMDYtYjljOC1mNjE5NzhlYzg1NDY=)


**Option 1 - Using Calculated Fields**

You make the value of of the secondary drop-down calculated based on the value of the first drop-down:
1. Open the settings for the secondary drop-down and naviagate to the data tab
2. Scroll down to the "Calculate Value" property and add a JavaScript expression to set the value based on the response to the first date/time field. The JavaScript expression below can be adapted fow your own specific form.

    ```javascript
    value = data.firstDateTime
    ```
3. Also in the data tab enable the "Allow Manual Override of Calculate Value" option. This will allow users to manually change the automatically calcualted value.



**Option 2 - Using the OnChange property**

Using the OnChange property of the first date/time field you can set the value of the second date/time. 

1. Open the component settings for the first date/time field and navigate to the logic tab
2. Use JavaScript in the onChange property to set the value of the second date/time field. Your expression will look something like this:

    ```javascript
    instance.root.getComponent("secondDateTime").setValue(data.firstDateTime);
    ```

**Reply by SSDK-Hello** [reply]
Perfect, thank you
